feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/sendrequest/SendRequestFragment.java

```diff
@@ -94,72 +99,99 @@ public void onClick(View view) {
+                                                                followRequest_ok = followRequest;
+                                                                requesterUsernames = followRequest.getRequesterUsernames();
+                                                                requesterUsernames.add(myUserID);
+                                                                DAO.createOrUpdate(recipientUsername, followRequest_ok, new VoidCallback() {
+                                                                    @Override
+                                                                    public void onCallback() {
+                                                                    }
+                                                                });
+                                                                SuccessfulSend();
+                                                            }
+
+                                                        }
+                                                    },null);
+                                                }
+                                            }
+                                        },null);
                             }
 
                         }
                     }, new VoidCallback() {
                         @Override
                         public void onCallback() {
-                            Toast toast = Toast.makeText(getActivity(), "User Not Exist, Invite your friend", Toast.LENGTH_LONG);
-                            toast.setGravity(Gravity.CENTER| Gravity.CENTER_HORIZONTAL, 0, 0);
-                            toast.show();
-                            usernameEditText.setText("");
+                            IndvalidUser();
                         }
                     });
+
                 }
             }
         });
         return root;
     }
 
+    public void AlreadyFollowedUser(){
+        Toast toast = Toast.makeText(getActivity(), "You have already followed this user", Toast.LENGTH_LONG);
+        toast.setGravity(Gravity.CENTER | Gravity.CENTER_HORIZONTAL, 0, 0);
+        toast.show();
+        usernameEditText.setText("");
+    }
 
-    /**
-     * Since Login/Register Activity haven't complete, so I use <filename> that I create manually. (which is Xiaole)
-     * Once sharedPreference create in the Login/Register Activity, I need to get the variable <file name>
-     * from mainActivity and pass the file name in SendRequestFragment --> TODO
-     *
-     * Login/Register should pass the filename variable to the MainActivity(Home Page).
-     */
+    public void AlreadySentRequest(){
+        Toast toast = Toast.makeText(getActivity(), "You have already sent request!", Toast.LENGTH_LONG);
+        toast.setGravity(Gravity.CENTER | Gravity.CENTER_HORIZONTAL, 0, 0);
+        toast.show();
+        usernameEditText.setText("");
+    }
 
+    public void ValidUser(){
```

Comment:
All of these new methods should be private, and ideally, be a bit more descriptive (ie. rename it to something like "showValidUserToast") or something